File: src/main/java/com/github/bin/entity/master/RoomRole.java
package com.github.bin.entity.master;

import com.baomidou.mybatisplus.annotation.TableField;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author bin
 * @since 2023/08/22
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RoomRole {
    private int id;

    @TableField("name")
    private String name;

    @TableField("color")
    private String color;

    public RoomRole copy(int id) {
        return new RoomRole(id, name, color);
    }

    @Override
    public String toString() {
        return "RoomRole(" +
                "id=" + id +
                ')';
    }
}

File: src/main/kotlin/com/github/bin/entity/RoomRole.kt
package com.github.bin.entity

import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName

@TableName("room_role", resultMap = "BaseResultMap")
class RoomRole() {
    @TableId("id", type = IdType.AUTO)
    var id: Int? = null

    @TableField("name")
    var name: String? = null

    @TableField("color")
    var color: String? = null

    constructor(id: Int, name: String, color: String) : this() {
        this.id = id
        this.name = name
        this.color = color
    }
}
